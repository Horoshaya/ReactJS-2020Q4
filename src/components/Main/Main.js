import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import MovieList from '../MovieList/MovieList';
import { FiltersSortBar } from '../FiltersSortBar/FiltersSortBar';
import { getMovie, sortByGenre } from '../../store/actions/actionCreators';

export const MainContext = React.createContext();

const Main = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMovie());
  }, [dispatch]);

  const sortMovieByGenre = (genre) => {
    dispatch(sortByGenre(genre, movieReducer));
  };

  const movieReducer = useSelector((store) => store.movieReducer);

  return (
    <main>
      <MainContext.Provider value={sortMovieByGenre}>
        <FiltersSortBar />
      </MainContext.Provider>
      <MovieList movieReducer={movieReducer} />
    </main>
  );
};

export default Main;
