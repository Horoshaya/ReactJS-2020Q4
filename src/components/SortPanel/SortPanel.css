.list {
  position: absolute;
  top: 35px;
  padding: 0;
  list-style: none;
  background: #555555;
  z-index: 15;
  width: 100%;
}

.listItem {
  padding: 15px 23px;
  font-size: 14px;
  color: #ffffff;
  text-transform: uppercase;
  background: none;
  border: none;
  cursor: pointer;
  outline: none;
}

.listItem:hover {
  background: #b83e48;
}

.dropdownHeader {
  color: #ffffff;
  display: inline-block;
}

.arrowUp:after {
  content: '\25B2';
  padding-left: 10px;
}

.arrowDown:after {
  content: '\25BC';
  padding-left: 10px;
}

.title,
.headerTitle {
  padding: 20px 15px;
  margin: 0;
  min-width: 150px;
  color: #ffffff;
  text-transform: uppercase;
  font-size: 18px;
  background: none;
  border: none;
  cursor: pointer;
  outline: none;
  display: inline-block;
}

.title {
  min-width: 50px;
}

.wrapper {
  position: relative;

  display: flex;
  justify-content: space-between;
}

.selected {
  background: #f65261;
}
