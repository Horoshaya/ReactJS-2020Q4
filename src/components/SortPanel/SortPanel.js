import React, { memo, useState, useContext } from 'react';
import classNames from 'classnames/bind';

import styles from './SortPanel.css';
import { usePropsLogger } from '../../hooks/usePropsLogger';
import { MainContext } from '../Main/Main';

let cx = classNames.bind(styles);

const InnerSortPanel = ({ headerTitle, list, toggleItem }) => {
  const [listOpen, setListOpen] = useState(false);
  const { sortByDateAndRatingHandle } = useContext(MainContext);

  const dropdaownClasses = cx({
    dropdownHeader: true,
    arrowUp: listOpen,
    arrowDown: !listOpen,
  });

  // const itemClasses = cx({
  //   listItem: true,
  //   selected: item.selected,
  // });

  const itemClasses = (item) =>
    cx({
      listItem: true,
      selected: item.selected,
    });

  const toggleList = () => {
    setListOpen(!listOpen);
  };

  usePropsLogger({ headerTitle, list, toggleItem }, 'SortPanel Props');

  return (
    <div className={styles.wrapper}>
      <p className={styles.title}>sort by</p>

      <div className={dropdaownClasses} onClick={toggleList}>
        <div className={styles.headerTitle}>{headerTitle}</div>
      </div>

      {listOpen && (
        <ul className={styles.list}>
          {list.map((item) => (
            <li
              className={itemClasses(item)}
              key={item.id}
              onClick={() => {
                toggleItem(item.id, item.key);
                sortByDateAndRatingHandle(item.title);
              }}
            >
              {item.title}
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export const SortPanel = memo(InnerSortPanel);
