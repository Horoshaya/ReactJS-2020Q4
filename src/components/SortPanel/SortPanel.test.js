import React from 'react';
import rerender from 'react-test-renderer';
import SortPanel from './SortPanel';
import { MainContext } from '../Main/Main';

const Foo = SortPanel.type;

describe('when SortPanel', () => {
  test('then snapshot created', () => {
    const component = rerender.create(
      <MainContext.Provider value={'test'}>
        <Foo list={[]} />
      </MainContext.Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
