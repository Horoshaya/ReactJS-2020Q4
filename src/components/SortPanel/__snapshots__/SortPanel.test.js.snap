// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when SortPanel then snapshot created 1`] = `
<div>
  <p>
    sort by
  </p>
  <div
    className="dropdownHeader arrowDown"
    onClick={[Function]}
  >
    <div />
  </div>
</div>
`;
